File: src/Http.API/ClientApp/src/app/share/admin/services/user.service.ts
import { Injectable } from '@angular/core';
import { BaseService } from './base.service';
import { Observable } from 'rxjs';
import { UserFilterDto } from '../models/user/user-filter-dto.model';
import { UserAddDto } from '../models/user/user-add-dto.model';
import { UserUpdateDto } from '../models/user/user-update-dto.model';
import { UserItemDtoPageList } from '../models/user/user-item-dto-page-list.model';
import { User } from '../models/user/user.model';

/**
 * 用户账户
 */
@Injectable({ providedIn: 'root' })
export class UserService extends BaseService {
  /**
   * 筛选
   * @param data UserFilterDto
   */
  filter(data: UserFilterDto): Observable<UserItemDtoPageList> {
    const url = `/api/admin/User/filter`;
    return this.request<UserItemDtoPageList>('post', url, data);
  }

  /**
   * 新增
   * @param data UserAddDto
   */
  add(data: UserAddDto): Observable<User> {
    const url = `/api/admin/User`;
    return this.request<User>('post', url, data);
  }

  /**
   * 更新
   * @param id 
   * @param data UserUpdateDto
   */
  update(id: string, data: UserUpdateDto): Observable<User> {
    const url = `/api/admin/User/${id}`;
    return this.request<User>('put', url, data);
  }

  /**
   * 详情
   * @param id 
   */
  getDetail(id: string): Observable<User> {
    const url = `/api/admin/User/${id}`;
    return this.request<User>('get', url);
  }

  /**
   * ⚠删除
   * @param id 
   */
  delete(id: string): Observable<User> {
    const url = `/api/admin/User/${id}`;
    return this.request<User>('delete', url);
  }

  /**
   * 修改密码
   * @param password 
   */
  changeMyPassword(password?: string): Observable<boolean> {
    const url = `/api/admin/User/password?password=${password}`;
    return this.request<boolean>('put', url);
  }

}

File: src/Sdks/Dusi.Manage.Client/Services/UserService.cs
using Share.Models.UserDtos;
namespace Dusi.Manage.Client.Services;
/// <summary>
/// 用户账户
/// </summary>
public class UserService : BaseService
{
    public UserService(HttpClient httpClient) : base(httpClient)
    {

    }
    /// <summary>
    /// 筛选
    /// </summary>
    /// <param name="data">UserFilterDto</param>
    /// <returns></returns>
    public async Task<PageList<UserItemDto>?> FilterAsync(UserFilterDto data)
    {
        string url = $"/api/admin/User/filter";
        return await PostJsonAsync<PageList<UserItemDto>?>(url, data);
    }

    /// <summary>
    /// 新增
    /// </summary>
    /// <param name="data">UserAddDto</param>
    /// <returns></returns>
    public async Task<User?> AddAsync(UserAddDto data)
    {
        string url = $"/api/admin/User";
        return await PostJsonAsync<User?>(url, data);
    }

    /// <summary>
    /// 更新
    /// </summary>
    /// <param name="id"> </param>
    /// <param name="data">UserUpdateDto</param>
    /// <returns></returns>
    public async Task<User?> UpdateAsync(string id, UserUpdateDto data)
    {
        string url = $"/api/admin/User/{id}";
        return await PutJsonAsync<User?>(url, data);
    }

    /// <summary>
    /// 详情
    /// </summary>
    /// <param name="id"> </param>
    /// <returns></returns>
    public async Task<User?> GetDetailAsync(string id)
    {
        string url = $"/api/admin/User/{id}";
        return await GetJsonAsync<User?>(url);
    }

    /// <summary>
    /// ⚠删除
    /// </summary>
    /// <param name="id"> </param>
    /// <returns></returns>
    public async Task<User?> DeleteAsync(string id)
    {
        string url = $"/api/admin/User/{id}";
        return await DeleteJsonAsync<User?>(url);
    }

    /// <summary>
    /// 修改密码
    /// </summary>
    /// <param name="password"> </param>
    /// <returns></returns>
    public async Task<bool?> ChangeMyPasswordAsync(string? password)
    {
        string url = $"/api/admin/User/password?password={password}";
        return await PutJsonAsync<bool?>(url);
    }

}